debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java

```diff
@@ -435,23 +435,20 @@ protected ChangeRecordEmitter getChangeRecordEmitter(SnapshotContext<O> snapshot
 
     /**
      * Returns a comma-separated list of columns to be used in the snapshot select.
-     * The selected columns are based on the column include/exclude filters and if all columns are excluded
      *
      * @param tableId the table to generate a query for
      * @return comma-separated list of columns
      */
     protected abstract String getSnapshotSelectColumns(TableId tableId);
 
     /**
-     * Returns a list of columns to be used in the snapshot select.
+     * Prepares a list of columns to be used in the snapshot select.
      * The selected columns are based on the column include/exclude filters and if all columns are excluded,
      * the list will contain all the primary key columns.
      *
-     * @param table the table metadata for a query
-     * @param tableId the table to generate a query for
      * @return list of snapshot select columns
      */
-    protected List<String> prepareSnapshotSelectColumns(Table table, TableId tableId) {
+    protected List<String> getPreparedColumnNames(Table table, TableId tableId, ColumnNamesResolver columnNameResolver) {
```

Comment:
This method should end with `return columnNameResolver.prepare(columnNames, tableId)`.  Otherwise the resolver is not being used at all currently.